docs/development/testguide.rst

```diff
@@ -143,63 +160,65 @@ Astropy Test Function
+declared in the package, before running the tests. This can therefore catch
+issues related to undeclared package data, or missing dependencies. Since we use
+tox to run many of the tests on continuous integration services, it can also be
+used in many cases to reproduce issues seen on those services.
 
-In contrast to the case of running from ``setup.py``, the ``--doctest-plus``
-and ``--doctest-rst`` options are not enabled by default when running the
-``pytest`` command directly. This flags should be explicitly given if they are
-needed.
+To run the tests with tox, first make sure that tox is installed, e.g.::
 
-Test-running options
-====================
+    pip install tox
 
-Running parts of the test suite
--------------------------------
+then run the basic test suite with::
 
-It is possible to run only the tests for a particular subpackage or set of
-subpackages.  For example, to run only the ``wcs`` tests from the
-commandline::
+    tox -e test
 
-    python setup.py test -P wcs
+or run the test suite with all optional dependencies with::
 
-Or, to run only the ``wcs`` and ``utils`` tests::
+    tox -e test-all
 
-    python setup.py test -P wcs,utils
+You can see a list of available test environments with::
 
-Or from Python::
+    tox -l
 
-    >>> import astropy
-    >>> astropy.test(package="wcs,utils")
+You can also run checks or commands not directly related to tests - for instance::
 
-You can also specify a single file to test from the commandline::
+    tox -e style
 
-    python setup.py test -t astropy/wcs/tests/test_wcs.py
+will run checks using the flake8 tool.
 
-When the ``-t`` option is given a relative path, it is relative to the
-installed root of astropy.  When ``-t`` is given a relative path to a
-documentation ``.rst`` file to test, it is relative to the root of the
-documentation, i.e. the ``docs`` directory in the source tree.  For
-example::
+Note that when running tests with ``tox``, it is not possible with the provided
+configuration to run just part of the test suite - for this we recommend running
+the tests as described in :ref:`running-pytest`. However, if is possible to pass
+options to pytest - to do this, add a ``--`` after the regular tox command, and
+anything after this will be passed to pytest, e.g.::
 
-    python setup.py test -t units/index.rst
+    tox -e test -- -v --pdb
+
+Test-running options
+====================
 
 .. _open-files:
 
 Testing for open files
 ----------------------
 
-Astropy can test whether any of the unit tests inadvertently leave any
-files open.  Since this greatly slows down the time it takes to run
-the tests, it is turned off by default.
+Using the :ref:`openfiles-plugin` plugin (which is installed automatically
+when installing pytest-astropy),  we can test whether any of the unit tests
+inadvertently leave any files open.  Since this greatly slows down the time it
+takes to run the tests, it is turned off by default.
```

Comment:
Ah, now I see why we do not do it by default in travis - sorry!

Reply:
Ah just saw this comment :) Marking as resolved.